src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs

```diff
@@ -257,7 +261,25 @@ protected override async Task<RunResult> RunAsyncInternal(string expr, string se
 
                 var newValue = await check.GetEvaluator().EvalAsync(CancellationToken.None, rtConfig);
 
-                return new RunResult(newValue);
+                // UntypedObjectType type is currently not supported for serialization.
+                if (newValue.Type is UntypedObjectType)
+                {
+                    return new RunResult(newValue);
+                }
+
+                // Get serialized expression. DateTimeValue needs timezone information.
+                var serializedExpression = newValue is DateTimeValue dtv ? dtv.ToExpression(iSetup.TimeZoneInfo) : newValue.ToExpression();
+
+                // Serialization test. Serialized expression must produce an identical result.
+                var checkSerialized = engine.Check(serializedExpression, parameters.Type, options: iSetup.Flags.ToParserOptions());
+                if (!check.IsSuccess)
+                {
```

Comment:
How could this fail?
Why do we need to set parser options?

Reply:
The idea is to run the serialized expression under the very same scenario when the original eval was performed.